node/src/components/contract_runtime/operations.rs

```diff
@@ -30,6 +32,50 @@ use casper_execution_engine::{
     storage::global_state::{CommitProvider, StateProvider},
 };
 
+fn generate_range_by_index(
+    highest_era: u64,
+    batch_size: u64,
+    batch_index: u64,
+) -> Option<Range<u64>> {
+    let start = batch_index.checked_mul(batch_size)?;
+    let end = cmp::min(start.checked_add(batch_size)?, highest_era);
+    Some(start..end)
+}
+
+/// Calculates era keys to be pruned.
+///
+/// Outcomes:
+/// * Ok(Some(range)) -- these keys should be purged
+/// * Ok(None) -- nothing to do, either done, or there is not enough eras to purge
+fn calculate_purge_eras(
```

Comment:
```suggestion
fn calculate_prune_eras(
```
It seems like we prefer this process to be called "pruning", which can be seen in [other places](https://github.com/casper-network/casper-node/pull/3811/files#diff-7884e55a78e0ad8ddf73a567cf9e82cb9ff732dfa408959a7140a5444d9f3b55R470) already.

And if not, then change "prune" to "purge" in other places (for consistency).

Reply:
@Fraser999 Thoughts on renaming? There's also chainspec entry `purge_batch_size` to change which we've already communicated to test

Reply:
I think Ed's preference was also "prune", so maybe worth changing now before the PR merges.

Reply:
Looks like you renamed it to prune everywhere except this file! :smile:

Reply:
Of course I forgot 🤦 Fixed